source/fundamentals/authentication/x509.txt

```diff
@@ -0,0 +1,129 @@
+.. _csharp-authentication-x509:
+
+=====
+X.509
+=====
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: .NET, certificate, code example
+
+Overview
+--------
+
+In the **X.509** authentication mechanism, the server and client use the
+:wikipedia:`TLS <Transport_Layer_Security>` protocol to exchange X.509 public-key
+certificates. You can use this mechanism to authenticate to MongoDB Atlas, MongoDB
+Enterprise Advanced, and MongoDB Community Edition.
+
+.. tip:: X.509 Mechanism
+
+   To learn how to use TLS/SSL with the {+driver-short+},
+   see :ref:`TLS/SSL <csharp-tls>`.
+
+   For more information about X.509 certificates, see
+   :ref:`X.509 <x509-client-authentication>` in the {+mdb-server+} manual.
+
+Code Placeholders 
+~~~~~~~~~~~~~~~~~
+
+The code examples on this page use the following placeholders:
+
+- ``<hostname>``: The network address of your MongoDB deployment.
+- ``<port>``: The port number of the MongoDB deployment. If you omit this parameter,
+  the driver uses the default port number (``27017``).
+- ``<X.509 certificate username>``: The username of the user associated with the X.509
+  certificate. The username must match the distinguished subject name of the certificate.
+  If you omit this parameter, the MongoDB deployment infers the username
+  from the X.509 certificate.
+- ``<path to X.509 certificate>``: The path to the X.509 certificate file.
+- ``<X.509 certificate password>``: The password for the X.509 certificate.
+
+To use the code examples on this page, replace these placeholders with your own values.
+
+Using X.509 Authentication in Your Application
+----------------------------------------------
+
+To use an X.509 certificate for authentication, you must use a
+``MongoClientSettings`` object to specify
+the path to your X.509 certificate and the password for the certificate. The certificate
+must use the :wikipedia:`PKCS #12<PKCS_12>` (``.p12``) or
+:wikipedia:`PEM<Privacy-Enhanced_Mail>` (``.pem``) format.
+
+You must also specify ``MONGODB-X509`` as the authentication mechanism. You can specify
+this mechanism either in your connection string or by using a ``MongoCredential`` object. 
+Select the :guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax:
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: default-connection-string
+
+      .. code-block:: csharp
+
+         var connectionString = "mongodb://<hostname>[:<port>]/?authMechanism=MONGODB-X509";
```

Comment:
Note to tech reviewer: Rishit got a timeout exception when using this code sample to connect to his deployment. Is the code sample correct?

Reply:
The time-out problem related to wrong connection string (it should be `mongo+srv` for Atlas)

Reply:
I've provided list of additional steps to successful connection to Atlas with x509 certificate via Slack, we can discuss the code sample if you want.